content/2017.03.building-projects-as-a-way-to-learn.rst

```diff
@@ -0,0 +1,258 @@
+ready experiments.
+
+The advantage we have at Mozilla, is that everything we do is
+published as Free, Libre, Opensource Software, which means that, even
+if, as Mozilla employees, we focus on something else, the work we've
+done is still available for the community to care about it.
+
+Thunderbird is an example of this. Mozilla stopped spending resources
+on it in July 2012 but five years later Mozilla Thunderbird is still
+actively used by a large number of users every day.
+
+When building software we solve complex issues and all the solutions
+we have put together are really useful for other people who might need
+to solve similar problems.
+
+For instance, Mozilla Persona lead the ground for `Firefox Accounts`_
+and the `Portier project`_, the Firefox Hello `loop server`_ shows a
+good example of how to build a WebRTC signaling server and was reused
+later by `Project Link`_. The `Remote Service Worker project`_ also
+shows how to use `Python asyncio`_ to handle a large pool of
+websockets_ using redis_ to route messages between them.
+
+.. _`Firefox Accounts`: https://developer.mozilla.org/en-US/docs/Mozilla/Tech/Firefox_Accounts
+.. _`Portier project`: https://portier.github.io/
+.. _`loop server`:  https://github.com/mozilla-services/loop-server
+.. _`Project Link`: https://wiki.mozilla.org/Connected_Devices/Projects/Project_Link
+.. _`Remote Service Worker project`: https://github.com/mozilla-services/remote-worker-server
+.. _`Python asyncio`: https://docs.python.org/3/library/asyncio.html
+.. _websockets: https://websockets.readthedocs.io/
+.. _redis: https://redis.io/
+
+
+Building software as a way to learn
+===================================
+
+.. image:: {filename}/images/conferences.jpg
+    :alt: Conference Room for DjangoCon Europe 2016
+    :align: center
+
+As you can see, every effort is not pointless because we learn a great
+deal from all these projects.
+
+Whether it takes us a month or a year to build something, we are
+learning. And if you want ot make the most value of this learning, it
+is important to share it as well.
+
+Luckily, there are at least one kind of people that you need to inform
+about what you are doing and learning about as part of your job: your
+colleagues.
+
+- You need to explain what you are trying to solve.
+- You need to find out how to solve the issue and agree on the
+  solutions you are putting together.
+- You need to take feedback to be able to choose the best solution
+  even before having written a single line of code.
+
+In a next article we will see how we make sharing the core of the
+process of building software as a distributed team.
+
+In the meantime, we can only encourage you to write about what you are
+building, go to conferences and tell others about what you learned and
+listen to them so that you don't have to go through the same pitfall.
+
+
+Conclusion
+==========
+
+In conclusion, don't build software to build software, build it to
```

Comment:
How about:
"don't build software to build software" => "don't build software just for the sake of it"